aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -264,15 +266,24 @@ private boolean isGlueIcebergTable(Table table) {
 
   @Override
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    TableOperations ops = newTableOps(identifier);
+    TableMetadata lastMetadata;
+    try {
+      lastMetadata = ops.current();
+    } catch (NoSuchKeyException ex) {
+      // S3 files already deleted, can not get S3 object through metadata location
+      lastMetadata = null;
+    }
     try {
-      TableOperations ops = newTableOps(identifier);
-      TableMetadata lastMetadata = ops.current();
       glue.deleteTable(DeleteTableRequest.builder()
           .catalogId(awsProperties.glueCatalogId())
           .databaseName(IcebergToGlueConverter.getDatabaseName(identifier))
           .name(identifier.name())
           .build());
       LOG.info("Successfully dropped table {} from Glue", identifier);
+      ValidationException.check(!purge || !awsProperties.glueLakeFormationEnabled() || lastMetadata == null,
+          "Can not delete table S3 files, should call deleteTable first " +
```

Comment:
The error message is a bit unclear from user perspective, my understanding is that "Cannot purge table with LakeFormation enabled because S3 access is lost after table is dropped. Please purge table data first before dropping the table"